src/widget/openglwindow.cpp

```diff
@@ -44,7 +37,10 @@ void OpenGLWindow::resizeGL(int w, int h) {
         // QGLWidget::resizeGL has devicePixelRatio applied, so we mimic the same behaviour
         m_pWidget->resizeGL(static_cast<int>(static_cast<float>(w) * devicePixelRatio()),
                 static_cast<int>(static_cast<float>(h) * devicePixelRatio()));
-        m_dirty = true;
+        // additional paint and swap to avoid flickering
+        m_pWidget->paintGL();
+        m_pWidget->swapBuffers();
```

Comment:
Doesn't trigger this a swap out of the VSync interval?
Shouldn't it be enough to paint to framebuffer with the new viewport size and wait for the next regular swap at VSync interval?

Reply:
Yes, it does trigger an additional swap for the widgets that are actually painting, but resizing messes with the swaps anyway. If you are talking about the case of `ST_PLL` it doesn't interfere with the syncing, because:

- The `frameSwapped` signal is only derived when Qt is handling the frame swap in response to an `update()` call, and not when the frame swap is done with an explicit `swapBuffers()` call` (at least, that is what I observe on macOS)
- The widget/OpenGLWindow that is driving the PLL is never resized.

> Shouldn't it be enough to paint to framebuffer with the new viewport size and wait for the next regular swap at VSync interval?

I can't get that to work without flickering.

Reply:
Now, maybe it is possible to skip the explicit paint & swap as triggered by vsync, if we detect that the resize already triggered it.